### PR TITLE
Faster handshake

### LINKED ISSUE #356: Use native hmac when possible
with PR #355 we are using a pure-python implementation of HMAC, that also means that it is slower than necessary

make test vectors to that code, use it only when necessary (as a fallback)

### PR DESCRIPTION
**Pull request by tomato42**
Optimise methods and functions used heavily in a handshake.

Speeds it up by about 25%

fixes #356

---
This change is [Reviewable](https://reviewable.io/reviews/tomato42/tlslite-ng/419)

### CODE REVIEW

File: tlslite/utils/codec.py [hunk 1]
```diff
@@ -324,6 +324,12 @@ def getFixBytes(self, lengthBytes):
         self.index += lengthBytes
         return bytes
 
+    def skip_bytes(self, length):
+        """Move the internal pointer ahead length bytes."""
+        if self.index + length > len(self.bytes):
+            raise DecodeError("Read past end of buffer")
```
Comment (Jakuje) [inline tlslite/utils/codec.py:331]: Maybe different exception message like "Skipped past the end of buffer"?

**Review by Jakuje — APPROVED** [review]
just minor comments. Looks good otherwise

File: tlslite/utils/codec.py (hunk 1 above)
Comment (tomato42) [reply tlslite/utils/codec.py:331]: this method is used only to get to the point we want to read, so while "skipped past..." would make more sense on the level of the Parser class, for the classes higher up the stack I think "Read past end of buffer" is more understandable

File: tlslite/utils/compat.py [hunk 2]
```diff
@@ -92,13 +120,23 @@ def compat26Str(x): return str(x)
...
+            return len(bin(val))-2
     else:
         def compat26Str(x): return x
 
         def remove_whitespace(text):
             """Removes all whitespace from passed in string"""
             return re.sub(r"\s+", "", text, flags=re.UNICODE)
 
+        def bit_length(val):
+            """Return number of bytes necessary to represent an integer."""
```
[the remark this answers is not in the document]
Comment (tomato42) [reply tlslite/utils/compat.py]: fixed

**Comment by tomato42** [comment]
thanks for the review!

**Pull request merged**